UI/Qt: Move New Tab button in tab bar

lmutter opened the pull request:
Uses the tab bar's width to determine the location of the button. The button is hidden if the window is small enough

This is in reference to #252

![image](https://github.com/user-attachments/assets/38f952a9-9d8c-4200-9a98-1e68bd531992)
![image](https://github.com/user-attachments/assets/f6d4894c-37c6-4122-a813-000ae1899987)
![image](https://github.com/user-attachments/assets/44cba7a7-c58c-4cf0-841b-454ebd4220bf)
![image](https://github.com/user-attachments/assets/da18743e-b7a3-4229-a2c6-a2dfb79f174f)

QToolBar with a single action is used instead of just a QToolButton, there is an issue where the button is completely black when just using a button. If that is resolved, it could be used instead.

circl-lastname commented on Ladybird/Qt/BrowserWindow.h:
```diff
@@ -212,6 +214,7 @@ public slots:
     StringView m_webdriver_content_ipc_path;
 
     bool m_allow_popups { false };
+    QToolBar* m_new_tab_button_toolbar { nullptr };
```
A bit of an awkward place to put this, it is up to you, but maybe it could be below `m_zoom menu` like so?
```cpp
    QMenu* m_zoom_menu { nullptr };

    QToolBar* m_new_tab_button_toolbar { nullptr };

    QMenu* m_hamburger_menu { nullptr };
```

lmutter commented:
Fixed the sizing so that it always makes room for the button, also fixed its position when the new menu bar is enabled

![image](https://github.com/user-attachments/assets/a304cbb4-d9e1-4ab8-88ec-2bddee93861b)
![image](https://github.com/user-attachments/assets/5e9c442d-3efd-45be-9e94-378c07b259ea)
![image](https://github.com/user-attachments/assets/0698f3ad-8755-4ab0-94c0-53a34495ea3d)
![image](https://github.com/user-attachments/assets/fa37bdd5-6bb8-40e9-a6e9-1894c28a41ee)

circl-lastname commented:
Thanks! This resolves my comment. Also. please squash these two commits together

ADKaster reviewed (approved):
This looks great! Let's get this in tree and let folks iterate on it if there's any other edge cases.
